### PR TITLE
JN-372, DSEC-659 include sonar in Java CI

### PR DESCRIPTION
**Pull request by TomConner**
Create new Sonar workflow to scan Java code in
* [juniperQuality Gate Status](https://sonarcloud.io/summary/new_code?id=broadinstitute_juniper),

and TypeScript code in 
* [juniper-ui-adminQuality Gate Status](https://sonarcloud.io/summary/new_code?id=broadinstitute_juniper-ui-admin)
* [juniper-ui-coreQuality Gate Status](https://sonarcloud.io/summary/new_code?id=broadinstitute_juniper-ui-core)
* [juniper-ui-participantQuality Gate Status](https://sonarcloud.io/summary/new_code?id=broadinstitute_juniper-ui-participant)

Because Sonar's Java and TypeScript scanning want to run separately, there are four separate SonarCloud projects that all correspond to this one GitHub repo.

### CODE REVIEW

**Review by devonbush — APPROVED** [review]

**Comment by TomConner** [comment]
I'm reworking to separate the different subproject scans so that SonarQube is happy.

**Comment by TomConner** [comment]
mark it *down* and a [link](google.com) too

**Comment by TomConner** [comment]
[Quality Gate Status](https://sonarcloud.io/summary/new_code?id=broadinstitute_juniper-ui-admin)

**Comment by TomConner** [comment]
[Quality Gate Status](https://sonarcloud.io/summary/new_code?id=broadinstitute_juniper-ui-admin)

**Comment by TomConner** [comment]
[Quality Gate Status](https://sonarcloud.io/summary/new_code?id=broadinstitute_juniper-ui-admin&branch=DSEC-659)

File: .github/workflows/sonar.yaml [hunk 1]
```diff
@@ -1,24 +1,67 @@
 name: Sonar
-on: workflow_dispatch
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - master
```
Comment (MatthewBemis) [inline .github/workflows/sonar.yaml:6]: Should this be triggered by pushes to the `development` branch instead? `development` is ultimately what ends up going to production, as we don't use a `master` or `main` branch.

**Review by MatthewBemis — APPROVED** [review]
Looks good to me. 👍  I just have one question about the workflow trigger.

File: .github/workflows/sonar.yaml (hunk 1 above)
Comment (TomConner) [reply .github/workflows/sonar.yaml:6]: Yes. Good catch.

**Pull request merged**